src/ReactWrapperComponent.jsx

```diff
@@ -54,7 +54,7 @@ export default function createWrapperComponent(node, options = {}) {
       const component = this._reactInternalInstance._renderedComponent;
       const inst = component.getPublicInstance();
       if (inst === null) {
-        return component;
+        return component._instance;
```

Comment:
at this point we're not really calling `getPublicInstance` at all. These lines could change to `return component._instance` perhaps (relevant commits: https://github.com/airbnb/enzyme/commit/d395726360a68694df1f36c40ef2e924169acca2, eg)

I'm wary of changing what appears to be the sole purpose of this method (to have different logic for SFCs). Could we change where findDOMnode is called instead? I'm not sure what would be best.